### PR TITLE
Match morphology loading and distance calculations

### PR DESCRIPTION
**Pull request by mgeplf**
* match them to internal BBP tools

### CODE REVIEW

File: bluepyopt/ephys/parameterscalers.py [hunk 1]
```diff
@@ -97,16 +97,17 @@ def scale(self, value, segment, sim=None):
         soma = segment.sec.cell().soma[0]
 
         # Initialise origin
-        sim.neuron.h.distance(0, 0.5, sec=soma)
+        sim.neuron.h.distance(0, 0, sec=soma)
```
Comment (wvangeit) [inline bluepyopt/ephys/parameterscalers.py]: Is this what Neurodamus does ? 
Mmm, need to discuss this with M. Hines, should probably be .5

File: bluepyopt/ephys/parameters.py [hunk 2]
```diff
@@ -228,7 +228,7 @@ def instantiate(self, sim=None, icell=None):
 
         for location in self.locations:
             for isection in location.instantiate(sim=sim, icell=icell):
-                for seg in isection:
+                for seg in isection.allseg():
```
Comment (wvangeit) [inline bluepyopt/ephys/parameters.py]: What exactly the difference here ? 
'for seg in section' is normally the advised way of doing this

Comment (wvangeit) [inline bluepyopt/ephys/parameters.py]: Mmm, ok: http://www.neuron.yale.edu/phpbb/viewtopic.php?f=2&t=2150
And Neurodamus is applying the 'allseg' behaviour ? We need to check if that's correct

Comment (mgeplf) [inline bluepyopt/ephys/parameters.py]: The HOC code is: `forsec sl for(x) { ... }` which is equivalent to allseg

Comment (wvangeit) [inline bluepyopt/ephys/parameters.py]: Does this change the output traces ?

File: bluepyopt/ephys/parameterscalers.py (hunk 1 above)
Comment (mgeplf) [inline bluepyopt/ephys/parameterscalers.py]: To get exact agreement between this and BBP, this is what I had to do.  For example, with the above, I get:
```
gIhbar_Ih (-0.869600 + 2.087000*math.exp(14.0017 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(21.9453 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(29.8889 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(29.8889 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(45.8799 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(61.8708 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(61.8708 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(71.6036 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(91.0691 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(110.535 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(130.0 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(149.466 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(168.931 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(188.397 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(207.862 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(227.328 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(237.061 * 0.003000)) * 0.000051
```

Which is exactly what I get when I dump the pre-exec code from the HOC file:
```
Cell[0].dend[0] gIhbar_Ih(0.000000) = (-0.869600 + 2.087000*exp(14.0017 * 0.003000)) * 0.000051
Cell[0].dend[0] gIhbar_Ih(0.500000) = (-0.869600 + 2.087000*exp(21.9453 * 0.003000)) * 0.000051
Cell[0].dend[0] gIhbar_Ih(1.000000) = (-0.869600 + 2.087000*exp(29.8889 * 0.003000)) * 0.000051
Cell[0].dend[1] gIhbar_Ih(0.000000) = (-0.869600 + 2.087000*exp(29.8889 * 0.003000)) * 0.000051
Cell[0].dend[1] gIhbar_Ih(0.500000) = (-0.869600 + 2.087000*exp(45.8799 * 0.003000)) * 0.000051
Cell[0].dend[1] gIhbar_Ih(1.000000) = (-0.869600 + 2.087000*exp(61.8708 * 0.003000)) * 0.000051
Cell[0].dend[2] gIhbar_Ih(0.000000) = (-0.869600 + 2.087000*exp(61.8708 * 0.003000)) * 0.000051
Cell[0].dend[2] gIhbar_Ih(0.055556) = (-0.869600 + 2.087000*exp(71.6036 * 0.003000)) * 0.000051
Cell[0].dend[2] gIhbar_Ih(0.166667) = (-0.869600 + 2.087000*exp(91.0691 * 0.003000)) * 0.000051
Cell[0].dend[2] gIhbar_Ih(0.277778) = (-0.869600 + 2.087000*exp(110.535 * 0.003000)) * 0.000051
Cell[0].dend[2] gIhbar_Ih(0.388889) = (-0.869600 + 2.087000*exp(130 * 0.003000)) * 0.000051
Cell[0].dend[2] gIhbar_Ih(0.500000) = (-0.869600 + 2.087000*exp(149.466 * 0.003000)) * 0.000051
Cell[0].dend[2] gIhbar_Ih(0.611111) = (-0.869600 + 2.087000*exp(168.931 * 0.003000)) * 0.000051
Cell[0].dend[2] gIhbar_Ih(0.722222) = (-0.869600 + 2.087000*exp(188.397 * 0.003000)) * 0.000051
Cell[0].dend[2] gIhbar_Ih(0.833333) = (-0.869600 + 2.087000*exp(207.862 * 0.003000)) * 0.000051
Cell[0].dend[2] gIhbar_Ih(0.944444) = (-0.869600 + 2.087000*exp(227.328 * 0.003000)) * 0.000051
Cell[0].dend[2] gIhbar_Ih(1.000000) = (-0.869600 + 2.087000*exp(237.061 * 0.003000)) * 0.000051
```

With the old code (ie: with the 0.5), I get:
```
gIhbar_Ih (-0.869600 + 2.087000*math.exp(0.0 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(7.94358 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(15.8872 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(15.8872 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(31.8781 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(47.8691 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(47.8691 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(57.6019 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(77.0674 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(96.533 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(115.998 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(135.464 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(154.93 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(174.395 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(193.861 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(213.326 * 0.003000)) * 0.000051
gIhbar_Ih (-0.869600 + 2.087000*math.exp(223.059 * 0.003000)) * 0.000051
```

File: bluepyopt/ephys/parameters.py (hunk 2 above)
Comment (mgeplf) [inline bluepyopt/ephys/parameters.py]: I can't say if that's 'correct' or if that's what the intended behaviour is.

**Comment by wvangeit** [comment]
I'm gonna close this because it looks like this well be fixed at the Neurodamus side

**Pull request closed without merging**